### DISCUSSION TITLE
Idea: Use smack as a Clang drop-in replacement

### DISCUSSION THREAD

**Discussion by teaalltr** [smackers/smack#732] [Ideas]
It would be nice to have smack as a clang drop-in replacement, so we'd just need to call smack with all the clang arguments plus the args to smack as arguments to the custom llvm pass.

Moreover, this way you could just set the env var CC or CXX and all would compile as usual, plus the memory checks and such.

**Comment by shaobo-he** [comment]
Hey @Piruzzolo, thank you for bringing it up. I tentatively moved it from issues to discussions. We'll use the latter for new ideas.
